### PR TITLE
Add support for non-function calling LLMs in llm graph transformers

### PR DESCRIPTION
**Pull request by tomasonjo**

### CODE REVIEW

File: libs/experimental/langchain_experimental/graph_transformers/llm.py [hunk 1]
```diff
@@ -99,6 +157,91 @@ class _Graph(BaseModel):
...
+IMPORTANT NOTES:
+- Don't add any explanation and text."""
+
+    system_message = SystemMessage(content=system_prompt)
+    parser = JsonOutputParser(pydantic_object=UnstructuredRelation)
+
+    human_prompt = PromptTemplate(
+        template="""Based on the following example, extract entities and 
+relations from the provided text.\n\n
+Use the following entity types, don't use other entity that is not defined below:
```
Comment (jexp) [inline libs/experimental/langchain_experimental/graph_transformers/llm.py:215]: should we add this conditionally to the prompt depending if we have node_labels and rel_types?

File: libs/experimental/langchain_experimental/graph_transformers/llm.py [hunk 2]
```diff
@@ -341,8 +501,21 @@ def process_response(self, document: Document) -> GraphDocument:
...
+            raw_schema = cast(Dict[Any, Any], raw_schema)
+            nodes, relationships = _convert_to_graph_document(raw_schema)
+        else:
+            nodes = []
+            relationships = []
+            parsed_json = self.json_repair.loads(raw_schema.content)
+            for rel in parsed_json:
+                source_node = Node(id=rel["head"], type=rel["head_type"])
+                target_node = Node(id=rel["tail"], type=rel["tail_type"])
+                relationships.append(
```
Comment (jexp) [inline libs/experimental/langchain_experimental/graph_transformers/llm.py:514]: where do we append nodes?

File: libs/experimental/langchain_experimental/graph_transformers/llm.py [hunk 3]
```diff
@@ -320,19 +463,36 @@ def __init__(
...
-        self.chain = prompt | structured_llm
+                self.json_repair = json_repair
+            except ImportError:
+                raise ImportError(
+                    "Could not import json_repair python package. "
+                    "Please install it with `pip install json-repair`."
+                )
+            self._function_call = False
+            # We do not allow overwriting unstructured prompt
+            unstructured_prompt = create_unstructured_prompt(
```
[the remark this answers is not in the document]
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:487]: We can allow that, but it has a predefined output which needs to be followed. We could think about something like additional instructions params that just adds some text to prompt.

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 1 above)
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:215]: sure

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 2 above)
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:514]: We don't because we don't really need them

**Comment by tomasonjo** [comment]
@baskaryan I've addressed the comments

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 2 above)
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:532]: But I added them anyway

**Comment by tomasonjo** [comment]
It would make sense to maybe add "additional_instructions" params, so that users don't have to rewrite the whole prompt, but that can wait for another PR

File: libs/experimental/langchain_experimental/graph_transformers/llm.py [hunk 4]
```diff
@@ -320,19 +475,39 @@ def __init__(
...
-        self.chain = prompt | structured_llm
+                self.json_repair = json_repair
+            except ImportError:
+                raise ImportError(
+                    "Could not import json_repair python package. "
+                    "Please install it with `pip install json-repair`."
+                )
+            self._function_call = False
+            # Allow passing prompts from users
+            if not prompt == default_prompt:
```
Comment (baskaryan) [inline libs/experimental/langchain_experimental/graph_transformers/llm.py:499]: nit: can we give `prompt` a default value of None so we can more rigorously check if user has passed in a value or not?

**Review by baskaryan — COMMENTED** [review]
one nit otherwise lgtm. will let @jexp also approve

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 4 above)
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py]: Fixed

**Comment by tomasonjo** [comment]
@jexp addressed all the comments. Waiting for an approval

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 2 above)
Comment (jexp) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:530]: thx, you never know what people will use them for :)

**Review by jexp — APPROVED** [review]
LGTM

File: libs/experimental/langchain_experimental/graph_transformers/llm.py [hunk 5]
```diff
@@ -317,22 +472,39 @@ def __init__(
...
-        schema = create_simple_model(allowed_nodes, allowed_relationships)
-        structured_llm = llm.with_structured_output(schema, include_raw=True)
-        self.chain = prompt | structured_llm
+                self.json_repair = json_repair
+            except ImportError:
+                raise ImportError(
+                    "Could not import json_repair python package. "
+                    "Please install it with `pip install json-repair`."
+                )
+            self._function_call = False
```
Comment (baskaryan) [inline libs/experimental/langchain_experimental/graph_transformers/llm.py:497]: nit: not needed

**Review by baskaryan — APPROVED** [review]

File: libs/experimental/langchain_experimental/graph_transformers/llm.py (hunk 5 above)
Comment (tomasonjo) [reply libs/experimental/langchain_experimental/graph_transformers/llm.py:497]: I didn't find the json-repair in poetry deps or lock?

**Pull request merged**